pyface/ui/qt4/data_view/data_view_widget.py

```diff
@@ -0,0 +1,65 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.api import Bool, Instance, observe, provides
+
+from pyface.qt.QtCore import QAbstractItemModel
+from pyface.qt.QtGui import QTreeView
+from pyface.data_view.abstract_data_model import AbstractDataModel
+from pyface.data_view.i_data_view_widget import (
+    IDataViewWidget, MDataViewWidget
+)
+from pyface.ui.qt4.widget import Widget
+from .data_view_item_model import DataViewItemModel
+
+# XXX This file is scaffolding and may need to be rewritten
+
+
+@provides(IDataViewWidget)
+class DataViewWidget(MDataViewWidget, Widget):
+    """ The Qt implementation of the DataViewWidget. """
+
+    #: The QAbstractItemModel instance used by the view.  This will
+    #: usually be a DataViewItemModel subclass.
+    _item_model = Instance(QAbstractItemModel)
+
+    def _create_control(self, parent):
+        """ Create the DataViewWidget's toolkit control. """
+        self._create_item_model()
+
+        control = QTreeView(parent)
+        control.setUniformRowHeights(True)
+        control.setModel(self._item_model)
+        control.setHeaderHidden(not self.header_visible)
+        return control
+
+    def _create_item_model(self):
+        """ Create the DataViewItemModel which wraps the data model. """
+        self._item_model = DataViewItemModel(self.data_model)
+
+    def destroy(self):
+        """ Perform any actions required to destroy the control.
+        """
+        super().destroy()
+        # ensure that we release the reference to the item model
+        self._item_model = None
+
```

Comment:
The correct approach here is to call `self.control.setModel(None)` before calling `super().destroy()`.  We don't want to be creating a new, transient, spurious data model.  `self._item_model.deleteLater()` is redundant, and may be bad if we end up allowing multiple widgets to share the same item model instance (which is allowed - even encouraged - by Qt).

Reply:
Ah I was looking into similar issues in TraitsUI's ImageEnumEditor where the model was kept alive and was responding to events even after the editor is finished.  `QComboBox.setModel` does not allow `None` to be set, and the rest of the internet recommended setting an empty model. Looks like `QTreeView` is fine with the model being set `None`. So yes I agree that would be preferred.

As for `deleteLater`, since the item model is created and owned by `DataViewWidget`, I think it makes sense for `DataViewWidget` to be responsible for deleting the model. If the item model is shared by multiple views, given the item model is private, the views should be children views managed by `DataViewWidget` already. In other words, `DataViewWidget` has the control.

One scenario where `DataViewWidget` should not delete the item model is if the item model isn't created by it but passed in externally, but I think we don't have that scenario here.

Reply:
The more important part of this is resolved: The model is removed from the widget so it can be silenced even if it sticks around.

The `deleteLater` is less important given that the model should now be silenced, although I still think it would be correct to call it explicitly because this `DataViewWidget` owns it and therefore responsible for its cleanup. 

Resolving.